test/diff/line-number-test

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+. libtest.sh
+. libgit.sh
+
+export COLUMNS=40
+export LINES=3
+
+steps '
+	:view-diff
+	:move-last-line
+	:move-up
+	:!echo %(lineno)
+	:save-display added-line-number.screen
+'
+
+
+git_init "$work_dir"
+
+test_setup_work_dir()
+{
+	git_commit --allow-empty-message
+	git_add README.md 'new line 1\n'
```

Comment:
Considering the definition of `git_add()` in `libgit.sh`, I must admit I don't understand how the test can work... It does, though.

For clarity, what about:

```patch
diff --git a/test/diff/line-number-test b/test/diff/line-number-test
index 72dac70..c44873e 100755
--- a/test/diff/line-number-test
+++ b/test/diff/line-number-test
@@ -9,18 +9,17 @@ export LINES=3
 steps '
        :view-diff
        :move-last-line
-       :move-up
        :!echo %(lineno)
        :save-display added-line-number.screen
 '

-
 git_init "$work_dir"

 test_setup_work_dir()
 {
        git_commit --allow-empty-message
-       git_add README.md 'new line 1\n'
+       printf 'new line 1\n' > README.md
+       git add README.md
 }
```

Reply:
I agree that this a bad abstraction; will change it (also I hadn't realized there was an extra newline).

`git_add()` calls `file()` to create the file. `file()` is defined in `libtest.sh`. That's really surprising because `file` usually means something else.

Reply:
Thanks, I missed file() [very bad idea IMHO] so git_add() looked a bit obscure but it makes sense now. Actually it was not an extra newline but git_add didn't interpret \n and so there was no new line at all. git show complaining about the lack of new line was the cause of the second line.